searx/network/__init__.py

```diff
@@ -53,14 +53,14 @@ def set_timeout_for_thread(timeout, start_time=None):
 
 
 def set_context_network_name(network_name):
-    THREADLOCAL.network = get_network(network_name)
+    THREADLOCAL.__dict__['network'] = get_network(network_name)
```

Comment:
Using `__dict__`  is only needed in _getters_ when you want None if unset, but should not used in _setters_.